File: public/dsa/arrays/pascals_triangle/combinatorics/pascals_triangle.py
import math
from typing import List


class Solution:
    def pascalTriangleRow(self, rowNo: int) -> List[int]:
        n = rowNo - 1
        return [math.comb(n, r) for r in range(n + 1)]


# tests-start
rowNo = 5
sol = Solution()
print(sol.pascalTriangleRow(rowNo))
# tests-end

'''output
[1, 4, 6, 4, 1]
'''

File: public/dsa/arrays/pascals_triangle/combinatorics/pascals_triangle.cpp

#include <bits/stdc++.h>
using namespace std;

vector<int> pascalTriangleRow(int rowNo) {
    int n = rowNo - 1;
    vector<int> row;
    for (int r = 0; r <= n; r++) {
        // Compute nCr
        int comb = 1;
        for (int i = 0; i < r; i++) {
            comb = comb * (n - i) / (i + 1);
        }
        row.push_back(comb);
    }
    return row;
}

// tests-start
int main() {
    int rowNo = 5;
    auto row = pascalTriangleRow(rowNo);
    for (int v : row) {
        cout << v << " ";
    }
    cout << endl;
    return 0;
}
// tests-end

/*output
1 4 6 4 1
*/